Add more logging to iOS print preview generation

rr-codes opened the pull request:
#### 6fbc6c232e8b153cf99f89483fea48716ba58bc4
<pre>
Add more logging to iOS print preview generation
<a href="https://bugs.webkit.org/show_bug.cgi?id=254377">https://bugs.webkit.org/show_bug.cgi?id=254377</a>
rdar://107159671

Reviewed by Tim Horton and Aditya Keerthi.

* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::computePageRects):
(WebCore::PrintContext::computePageRectsWithPageSize):
(WebCore::PrintContext::computePageRectsWithPageSizeInternal):
(WebCore::PrintContext::spoolPage):
* Source/WebCore/platform/Logging.h:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView _attributesForPrintFormatter:]):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::beginPrinting):
(WebKit::WebPage::endPrinting):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::computePagesForPrintingiOS):
(WebKit::WebPage::drawToImage):

Canonical link: <a href="https://commits.webkit.org/262097@main">https://commits.webkit.org/262097@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed5de3ee088517db02603d83ff0eb8539d3e4351

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/560 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/578 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/604 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/666 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/553 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/626 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/663 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/732 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/568 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/537 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/535 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/665 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/598 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/521 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/560 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/494 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/542 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/528 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/481 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/537 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/129 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/537 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/ios/WKContentView.mm:
```diff
@@ -997,6 +997,9 @@ - (BOOL)_wk_printFormatterRequiresMainThread
 
     auto attributes = adoptNS([[_WKPrintFormattingAttributes alloc] initWithPageCount:pageCount frameID:frameID printInfo:printInfo]);
+
+    RELEASE_LOG(Printing, "Computed attributes for print fomatter. Computed page count = %zu", pageCount);
```
"fomatter"

rr-codes replied:
Good catch!

pxlcoder commented on Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
```diff
@@ -4376,6 +4376,8 @@ static bool selectionIsInsideFixedPositionContainer(LocalFrame& frame)
     auto margin = printInfo.margin;
     computePagesForPrintingImpl(frameID, printInfo, pageRects, totalScaleFactor, margin);
 
+    RELEASE_LOG(Printing, "Computing pages for printing (iOS). Page rects size = %zu", pageRects.size());
```
`(iOS)` isn't really necessary, since we'll already know the logging is from iOS...

rr-codes committed:
Add more logging to iOS print preview generation
https://bugs.webkit.org/show_bug.cgi?id=254377
rdar://107159671

Reviewed by Tim Horton and Aditya Keerthi.

* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::computePageRects):
(WebCore::PrintContext::computePageRectsWithPageSize):
(WebCore::PrintContext::computePageRectsWithPageSizeInternal):
(WebCore::PrintContext::spoolPage):
* Source/WebCore/platform/Logging.h:
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView _attributesForPrintFormatter:]):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::beginPrinting):
(WebKit::WebPage::endPrinting):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::computePagesForPrintingiOS):
(WebKit::WebPage::drawToImage):

Canonical link: https://commits.webkit.org/262097@main
